python-avd/pyavd/_eos_designs/structured_config/base/__init__.py

```diff
@@ -164,64 +148,61 @@ def ipv6_static_routes(self) -> None:
             vrf=self.inputs.mgmt_interface_vrf, destination_address_prefix="::/0", gateway=self.shared_utils.ipv6_mgmt_gateway
         )
 
-    @cached_property
-    def service_routing_protocols_model(self) -> str:
+    @structured_config_contributor
+    def service_routing_protocols_model(self) -> None:
         """service_routing_protocols_model set to 'multi-agent'."""
-        return "multi-agent"
+        self.structured_config.service_routing_protocols_model = "multi-agent"
 
-    @cached_property
-    def ip_routing(self) -> bool | None:
+    @structured_config_contributor
+    def ip_routing_ipv6_interfaces(self) -> None:
+        """ip_routing_ipv6_interfaces set based on underlay_rfc5549 variable."""
+        if not self.shared_utils.underlay_router and not self.shared_utils.node_config.always_configure_ip_routing:
+            return
+
+        if self.inputs.underlay_rfc5549:
```

Comment:
let's move this check first as this is just checking inputs and would return immediately without having to check for the more complex ones above